common/src/chain/block/mod.rs

```diff
@@ -118,6 +118,31 @@ impl Block {
         Ok(block)
     }
 
+    // this function is needed to avoid a circular dependency with storage
```

Comment:
We should think about how the code base is organised into crates to avoid stuff like this.

Reply:
Agreed. Though keep in mind that the primary reason for not solving this now, apart from that it's not a big deal right now, is that Roy is working on the consensus struct.

Reply:
I'm resolving this because no action can be done right now and it's not a big deal.